src/net/p2p.rs

```diff
@@ -1131,23 +1131,27 @@ impl PeerNetwork {
 
         let mut registered = vec![];
 
-        for (event_id, client_sock) in poll_state.new.drain() {
-            // event ID already used?
-            if self.peers.contains_key(&event_id) {
-                continue;
-            }
-
-            match self.network {
+        for (hint_event_id, client_sock) in poll_state.new.drain() {
+            let event_id = match self.network {
                 Some(ref mut network) => {
                     // add to poller
-                    if let Err(_e) = network.register(self.p2p_network_handle, event_id, &client_sock) {
-                        continue;
+                    match network.register(self.p2p_network_handle, hint_event_id, &client_sock) {
+                        Ok(event_id) => event_id,
+                        Err(e) => {
+                            warn!("Failed to register {:?}: {:?}", &client_sock, &e);
+                            continue;
+                        }
                     }
                 },
                 None => {
                     test_debug!("{:?}: network not connected", &self.local_peer);
                     return Err(net_error::NotConnected);
                 }
+            };
+            
+            // event ID already used?
+            if self.peers.contains_key(&event_id) {
```

Comment:
This check used to occur before the call to `register` -- should that still be the case?

Reply:
Ah, it should also be deregistering the socket here as well.  This check can't happen before anymore, since `event_id` might not be equal to `hint_event_id`.

Reply:
Fixed in 986e210